sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/conditionalExpressions.scala

```diff
@@ -161,6 +161,13 @@ case class CaseWhen(
     elseValue: Option[Expression] = None)
   extends ComplexTypeMergingExpression with ConditionalExpression {
 
+  def this(branches: Seq[Expression]) = this(
```

Comment:
Does this need an additional test?

Reply:
let me see

Reply:
Can we use the `CaseWhen.createFromParser`? You will need to register an `ExpressionBuilder` for this. The upside of this is that it is less messy in the constructor, and that there is ample testing for this from the parser.

Reply:
@hvanhovell  what about we create CaseWhen in the [connect planner ](https://github.com/apache/spark/blob/master/connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala#L535-L554), then we don't need to change sql side at all.

Reply:
I think it is better to keep the connect planner as generic as possible. If we can use an unresolved function then it is better.

Reply:
oh, I mean we still use an unresolved function, and we can call `CaseWhen.createFromParser` in connect planner